src/components/coin.ts

```diff
@@ -72,23 +136,31 @@ export const updateCoinBalanceByUserId = async (userId: string, newBalance: numb
     updateAmount,
     updateAmount
   );
+  await createCoinLedgerEntry(userId, oldBalance, updateAmount, event, reason, adminId);
 };
 
 /*
-  If user doesn't exist, create a row with the specified amount as the balance.
-  Otherwise, adjust the user's balance by the specified amount.
-  The user's balance will be set to 0 if the adjustment brings it below 0.
+  Adds an entry to the Codey coin ledger due to a change in a user's coin balance.
+  reason is only applicable for admin commands and is optional.
+  adminId is only applicable for admin commands and is mandatory.
 */
-export const adjustCoinBalanceByUserId = async (userId: string, amount: number): Promise<void> => {
+export const createCoinLedgerEntry = async (
+  userId: string,
+  oldBalance: number,
+  updateAmount: number,
```

Comment:
nit: `updateAmount` confused me, I initially thought this is the amount diff or delta. Maybe rename to `newBalance`? Since we already have an `oldBalance` param here, which makes a lot of sense.